### PR TITLE
Fix async_let playground transform

### PR DESCRIPTION
**Pull request by etcwilde**
Async-let pattern-binding-decls can't be logged since they haven't been
await'd yet. This patch fixes it so the playground transform doesn't try
to log the async-let PBD.

Without this fix, you'll get an error like the following;
```
error: expression is 'async' but is not marked with 'await'
  async let asyncVar = asyncFn1()
            ^~~~~~~~
            await
```

rdar://85311228

### CODE REVIEW

**Comment by etcwilde** [comment]
@swift-ci please test

**Comment by etcwilde** [comment]
Failure looks unrelated
```
lldb version 13.0.0 (https://github.com/apple/llvm-project.git revision 099d1707e1a0530b4a35b14a711ff59f8ccb9db8)
Apple Swift version 5.7-dev (LLVM 099d1707e1a0530, Swift 384c75222e2a0f7)
Skipping the following test categories: ['frame-diagnose', 'watchpoint', 'dwo', 'swiftmaccatalyst', 'libstdcxx', 'dwo', 'llgs', 'fork']

--
Command Output (stderr):
--
FAIL: LLDB (/Users/ec2-user/jenkins/workspace/swift-PR-macos/branch-main/build/buildbot_incremental/llvm-macosx-x86_64/bin/clang-x86_64) :: test_dsym (lldbsuite.test.lldbtest.TestArrayBridgedEnum)
PASS: LLDB (/Users/ec2-user/jenkins/workspace/swift-PR-macos/branch-main/build/buildbot_incremental/llvm-macosx-x86_64/bin/clang-x86_64) :: test_dwarf (lldbsuite.test.lldbtest.TestArrayBridgedEnum)
UNSUPPORTED: LLDB (/Users/ec2-user/jenkins/workspace/swift-PR-macos/branch-main/build/buildbot_incremental/llvm-macosx-x86_64/bin/clang-x86_64) :: test_dwo (lldbsuite.test.lldbtest.TestArrayBridgedEnum) (test case does not fall in any category of interest for this run) 
UNSUPPORTED: LLDB (/Users/ec2-user/jenkins/workspace/swift-PR-macos/branch-main/build/buildbot_incremental/llvm-macosx-x86_64/bin/clang-x86_64) :: test_gmodules (lldbsuite.test.lldbtest.TestArrayBridgedEnum) (skipping (gmodules only makes sense for clang tests)) 
```
Hitting it again.

**Comment by etcwilde** [comment]
@swift-ci please test macOS

File: test/PlaygroundTransform/async_let.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    if x < 1 {
+        return 1
+    }
+    return await x * factorial(x - 1)
+}
+
+async let fac4 = factorial(4)
+print(await fac4)
+
+//             return await x * factorial(x - 1)
```
Comment (etcwilde) [inline test/PlaygroundTransform/async_let.swift:30]: Comment is wrong here, that should be `return 1`

**Review by DougGregor — APPROVED** [review]

**Pull request merged**